tests/fs/test_mmap_types.c

```diff
@@ -0,0 +1,103 @@
+#include <stdio.h>
+#include <sys/mman.h>
+#include <string.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <unistd.h>
+#ifdef EMSCRIPTEN
+#include <emscripten.h>
+#include <emscripten_mmap.h>
+#else
+#include "../../system/include/emscripten/emscripten_mmap.h"
+#endif
+
+#define define_writeread_test(T, name) \
```

Comment:
The test creates a file, writes it with mmap and then reads through mmap and through fread.

This is useful because before I normalized long -> uint32_t I got different results from fread/mmap on my 64-bit Apple machine.